test/gas.test.mjs

```diff
@@ -114,13 +103,21 @@ describe('Gas test', () => {
       );
 
       await web3Client.waitForEvent(eventLogs, ['blockProposed']);
+      const numberOfBlocksAfter = await countBlocksInOptimist();
+
+      expect(numberOfBlocksBefore).to.be.equal(numberOfBlocksAfter - 1);
+
+      const gasCostDeposit = await getLatestBlockGasUsed();
 
       const expectedGasCostPerTx = 100000 + 15000 * txPerBlock;
-      expect(gasCost).to.be.lessThan(expectedGasCostPerTx);
-      console.log('Deposit L1 average gas used was', averageL1GasCost(receipts));
+      expect(gasCostDeposit).to.be.lessThan(expectedGasCostPerTx);
+      logger.debug(`Deposit L1 average gas used was ${averageL1GasCost(receipts)}`);
 
-      await nf3Users[0].makeBlockNow();
-      await web3Client.waitForEvent(eventLogs, ['blockProposed']);
+      const mempoolTrancastions = await nf3Users[0].unprocessedTransactionCount();
+      if (mempoolTrancastions > 0) {
+        await nf3Users[0].makeBlockNow();
+        await web3Client.waitForEvent(eventLogs, ['blockProposed']);
+      }
```

Comment:
These `nf3` calls make requests to the optimist so it's best to use the Proposer instance (while still using `nf3` here and there...)

```suggestion
      const mempoolTransactions = await nf3Proposer1.unprocessedTransactionCount();
      if (mempoolTransactions > 0) {
        await nf3Proposer1.makeBlockNow();
        await web3Client.waitForEvent(eventLogs, ['blockProposed']);
      }
```

Reply:
And this could also be a util!! :pray:

Reply:
don't we want to make nf3Proposer1 from nf3 eventually obsolete, that is why i tend not to use it.

Reply:
I understand, but that doesn't mean that we should hide calls that belong to the proposer under the rug :sweat_smile: